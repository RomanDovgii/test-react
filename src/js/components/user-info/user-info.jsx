import React from "react";
import {connect} from "react-redux";

const UserInfo = ({selectedUser}) => {
  if (Object.keys(selectedUser).length === 0) {
    return null;
  } else {
    const {address, description} = selectedUser;

    return (
      <section className="user-info">
        <ul className="user-nfo__list">
          <li className="user-info__item">
            <p className="user-info__description">Выбран пользователь:</p>
            <p className="user-info__information">
              <b>{selectedUser.firstName} {selectedUser.lastName}</b>
            </p>
          </li>
          <li className="user-info__item user-info__item--description">
            <p className="user-info__description">Описание:</p>
            <textarea value={description} readOnly/>
          </li>
          <li className="user-info__item">
            <p className="user-info__description">Адрес проживания:</p>
            <p className="user-info__information">
              <b>{address.streetAddress}</b>
            </p>
          </li>
          <li className="user-info__item">
            <p className="user-info__description">Город:</p>
            <p className="user-info__information">
              <b>{address.city}</b>
            </p>
          </li>
          <li className="user-info__item">
            <p className="user-info__description">Провинция/Штат:</p>
            <p className="user-info__information">
              <b>{address.state}</b>
            </p>
          </li>
          <li className="user-info__item">
            <p className="user-info__description">Индекс:</p>
            <p className="user-info__information">
              <b>{address.zip}</b>
            </p>
          </li>
        </ul>
      </section>
    );
  }
};

const mapStateToProps = (state) => ({
  selectedUser: state.selectedUser
});

export {UserInfo};
export default connect(mapStateToProps)(UserInfo);
