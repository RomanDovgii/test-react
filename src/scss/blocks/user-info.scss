.user-info {
    margin-top: 30px;
    max-width: 50%;
    border: 3px solid #333333;
}

.user-nfo__list {
    list-style: none;
}

.user-info__item {
    display: flex;
    margin-top: 10px;

    &--description {
        flex-direction: column;
    }
}

.user-info__description {
    margin-right: 10px;
}
