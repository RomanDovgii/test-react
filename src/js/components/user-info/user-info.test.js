import React from "react";
import renderer from "react-test-renderer";
import {UserInfo} from "./user-info";

const mockUser = {
  id: 339,
  firstName: `Susen`,
  lastName: `Graham`,
  email: `LNicholson@mi.org`,
  phone: `(220)920-9427`,
  address: {
    streetAddress: `7422 Vitae Rd`,
    city: `Bergen County`,
    state: `NE`,
    zip: `24099`
  },
  description: `aenean augue neque mattis sollicitudin turpis sed aenean elit fringilla nec massa libero nullam lectus aenean et vel sed pulvinar et augue facilisis vitae libero dolor massa adipiscing dolor vitae orci tortor`
};

it(
    `Should render UserInfo`,
    () => {
      const tree = renderer
      .create(
          <UserInfo
            selectedUser={mockUser}
          />
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    }

);

it(
    `Shouldn't render UserInfo`,
    () => {
      const tree = renderer
      .create(
          <UserInfo
            selectedUser={{}}
          />
      )
      .toJSON();

      expect(tree).toMatchSnapshot();
    }
);
