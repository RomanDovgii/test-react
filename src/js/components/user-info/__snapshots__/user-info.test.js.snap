// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render UserInfo 1`] = `
<section
  className="user-info"
>
  <ul
    className="user-nfo__list"
  >
    <li
      className="user-info__item"
    >
      <p
        className="user-info__description"
      >
        Выбран пользователь:
      </p>
      <p
        className="user-info__information"
      >
        <b>
          Susen
           
          Graham
        </b>
      </p>
    </li>
    <li
      className="user-info__item user-info__item--description"
    >
      <p
        className="user-info__description"
      >
        Описание:
      </p>
      <textarea
        className="user-info__textarea"
        readOnly={true}
        value="aenean augue neque mattis sollicitudin turpis sed aenean elit fringilla nec massa libero nullam lectus aenean et vel sed pulvinar et augue facilisis vitae libero dolor massa adipiscing dolor vitae orci tortor"
      />
    </li>
    <li
      className="user-info__item"
    >
      <p
        className="user-info__description"
      >
        Адрес проживания:
      </p>
      <p
        className="user-info__information"
      >
        <b>
          7422 Vitae Rd
        </b>
      </p>
    </li>
    <li
      className="user-info__item"
    >
      <p
        className="user-info__description"
      >
        Город:
      </p>
      <p
        className="user-info__information"
      >
        <b>
          Bergen County
        </b>
      </p>
    </li>
    <li
      className="user-info__item"
    >
      <p
        className="user-info__description"
      >
        Провинция/Штат:
      </p>
      <p
        className="user-info__information"
      >
        <b>
          NE
        </b>
      </p>
    </li>
    <li
      className="user-info__item"
    >
      <p
        className="user-info__description"
      >
        Индекс:
      </p>
      <p
        className="user-info__information"
      >
        <b>
          24099
        </b>
      </p>
    </li>
  </ul>
</section>
`;

exports[`Shouldn't render UserInfo 1`] = `null`;
